src/server/pps/cmds/cmds.go

```diff
@@ -393,26 +394,47 @@ each datum.`,
 			if page < 0 {
 				return errors.Errorf("page must be zero or positive")
 			}
-			if raw {
+			var printF func(*ppsclient.DatumInfo) error
+			if !raw {
+				if output != "" {
+					cmdutil.ErrorAndExit("cannot set --output (-o) without --raw")
+				}
+				writer := tabwriter.NewWriter(os.Stdout, pretty.DatumHeader)
+				printF = func(di *ppsclient.DatumInfo) error {
+					pretty.PrintDatumInfo(writer, di)
+					return nil
+				}
+				defer func() {
+					if err := writer.Flush(); retErr == nil {
+						retErr = err
+					}
+				}()
+			} else {
 				e := encoder(output)
-				return client.ListDatumF(args[0], pageSize, page, func(di *ppsclient.DatumInfo) error {
+				printF = func(di *ppsclient.DatumInfo) error {
 					return e.EncodeProto(di)
-				})
-			} else if output != "" {
-				cmdutil.ErrorAndExit("cannot set --output (-o) without --raw")
+				}
 			}
-			writer := tabwriter.NewWriter(os.Stdout, pretty.DatumHeader)
-			if err := client.ListDatumF(args[0], pageSize, page, func(di *ppsclient.DatumInfo) error {
-				pretty.PrintDatumInfo(writer, di)
-				return nil
-			}); err != nil {
-				return err
+			if pipelineInputPath != "" {
+				pipelineReader, err := ppsutil.NewPipelineManifestReader(pipelineInputPath)
+				if err != nil {
+					return err
+				}
+				request, err := pipelineReader.NextCreatePipelineRequest()
+				if err != nil {
+					return err
+				}
+				return client.ListDatumInputF(request.Input, pageSize, page, printF)
+			} else if len(args) == 1 {
+				return client.ListDatumF(args[0], pageSize, page, printF)
+			} else {
```

Comment:
there needs to be a similar error case for if _both_ `pipelineInputPath != ""` and `len(args) == 1` or else one of the options is silently discarded